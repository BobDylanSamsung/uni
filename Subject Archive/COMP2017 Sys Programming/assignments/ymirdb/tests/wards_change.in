set a 1 2 3
set b 1 a 2
set c a b
set d a b c
set e 1 d 2 a 3 c 4 b 5
forward e
pluck e 2
pluck e 5
list entries
forward e
backward a
del e
list entries
forward a
forward b
forward c
forward d
backward a
backward b
backward c
backward d
set c 1 2 3 4
list entries
forward a
forward b
forward c
forward d
backward a
backward b
backward c
backward d
del d
del b
list entries
forward a
forward c
backward a
backward c
bye
