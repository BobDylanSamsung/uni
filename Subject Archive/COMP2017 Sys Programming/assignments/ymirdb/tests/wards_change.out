ok

ok

ok

ok

ok

a, b, c, d

d

c

e [1 2 a 3 4 b 5]
d [a b c]
c [a b]
b [1 a 2]
a [1 2 3]

a, b

b, c, d, e

ok

d [a b c]
c [a b]
b [1 a 2]
a [1 2 3]

nil

a

a, b

a, b, c

b, c, d

c, d

d

nil

ok

d [a b c]
c [1 2 3 4]
b [1 a 2]
a [1 2 3]

nil

a

nil

a, b, c

b, d

d

d

nil

ok

ok

c [1 2 3 4]
a [1 2 3]

nil

nil

nil

nil

bye
